PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -3980,21 +3983,21 @@ void ScintillaEditView::hideLines()
 		// Special case: user wants to hide every line in between other hidden sections.
 		// Both "while" loops are executed (merge with above AND below hidden section):
 
-		while (scope == 0)
+		while (scope == 0 && startMarker < nbLines)
 			removeMarker(--startMarker, 1 << MARK_HIDELINESBEGIN);

```

Comment:
How `--startMarker` can lead to condition `startMarker < nbLines` ??

Reply:
This could possibly happen if control flow enters `while` loop when there is an invalid state of the markers. In a nutshell, begin/end markers don't match -> this leads to `while` loops searching for the paired marker -> because paired marker is not found, `scope` variable is not changing and program freezes.
Variable `startMarker` can cause issues, because if there is end marker, but no begin marker, `startMarker` will keep decrementing (because it searches for the begin marker), starting from some line X, then 0, then overflow will occur. When we check `startMarker < nbLines`, then we basically can detect overflow.

Reply:
Then please use the type converting:
`while (scope == 0 && static_cast<intptr_t>(startMarker) < 0)`